### ISSUE TITLE
[BR]: dovecot jail causes 100% CPU usage

### ISSUE THREAD

**Issue by foutrelis** [fail2ban/fail2ban#3370]
### Environment:

- Fail2Ban version: 1.0.1-1
- OS, including release name/version: Arch Linux x86_64
- [X] Fail2Ban installed via OS/distribution mechanisms
- [X] You have not applied any additional foreign patches to the codebase
- [X] Some customizations were done to the configuration (provide details below is so)

### The issue:

`fail2ban-server` stays at 100% CPU usage and commands issued using `fail2ban-client` hang.

#### Steps to reproduce

Not sure yet, I suspect it is triggered by a logged Dovecot authentication failure.

#### Expected behavior

Low CPU usage and responsive `fail2ban-client`.

#### Observed behavior

Sudden 100% CPU usage a few hours after starting the fail2ban service. Restarting the service doesn't resolve it (I think it parses the same Dovecot log item that causes the 100% CPU usage in the first place).

#### Any additional information

Bisects to https://github.com/fail2ban/fail2ban/commit/dfc866ea410840a8c9bfba04d4fa92213221164d. Something about the new regular expression is triggering this behavior. It only happens on one of our ~38 servers after upgrading to fail2ban 1.0.1. The affected server is the only one with the `dovecot` jail enabled.

#### Any customizations done to /etc/fail2ban/ configuration

`findtime` set to 3600 and `maxretry` set to 8 ([nothing else changed](https://gitlab.archlinux.org/archlinux/infrastructure/-/blob/master/roles/fail2ban/templates/dovecot.jail.j2) I think)

#### Relevant parts of /var/log/fail2ban.log file:

Nothing relevant is logged.

#### Relevant lines from monitored log files:

I'll try to provide a repro case soon. I need to figure out which line is triggering the observed behavior.

**Comment by sebres** [comment]
Huh... unexpected by that RE.

> I'll try to provide a repro case soon. I need to figure out which line is triggering the observed behavior.

If it is some single line that causes that, you can try to do:
```
fail2ban-regex -l 5 -o msg /var/log/dovecot/dovecot-info.log dovecot | grep 'Working on line'
```
(replace `/var/log/dovecot/dovecot-info.log` with your path or `systemd-journal` if it is journal either)
And see which line may cause it (e. g. long pause, etc).

**Comment by foutrelis** [comment]
Thanks for the suggestion to use `fail2ban-regex`, it helped to find the triggering log lines with ease.

The following lines hang for me, although the third one (first `managesieve-login` at `03:04:08`) does get parsed successfully after 10-20 seconds:

```
Oct 04 01:49:33 mail.example.com dovecot[459]: imap-login: Disconnected: Connection closed: read(size=1026) failed: Connection reset by peer (no auth attempts in 0 secs): user=<>, rip=1.2.3.4, lip=1.2.3.4, TLS handshaking: read(size=1026) failed: Connection reset by peer
Oct 04 01:49:33 mail.example.com dovecot[459]: imap-login: Disconnected: Connection closed: SSL_accept() failed: error:1408F10B:SSL routines:ssl3_get_record:wrong version number (no auth attempts in 0 secs): user=<>, rip=1.2.3.4, lip=1.2.3.4, TLS handshaking: SSL_accept() failed: error:1408F10B:SSL routines:ssl3_get_record:wrong version number
Oct 04 03:04:08 mail.example.com dovecot[459]: managesieve-login: Disconnected: Too many invalid commands. (no auth attempts in 0 secs): user=<>, rip=1.2.3.4, lip=1.2.3.4
Oct 04 03:04:32 mail.example.com dovecot[459]: managesieve-login: Disconnected: Connection closed: read(size=1007) failed: Connection reset by peer (no auth attempts in 1 secs): user=<>, rip=1.2.3.4, lip=1.2.3.4
Oct 04 09:57:34 mail.example.com dovecot[472]: imap-login: Disconnected: Connection closed: SSL_accept() failed: error:14209102:SSL routines:tls_early_post_process_client_hello:unsupported protocol (no auth attempts in 0 secs): user=<>, rip=1.2.3.4, lip=1.2.3.4, TLS handshaking: SSL_accept() failed: error:14209102:SSL routines:tls_early_post_process_client_hello:unsupported protocol
```

**Comment by sebres** [comment]
OK, reproduced on some dovecot-log too - lines like that causes few seconds pause:
```
2022-10-04 10:31:49 imap-login: Info: Disconnected: Too many invalid commands (no auth attempts in 0 secs): user=<>, rip=192.0.2.111, lip=127.0.0.1, session=<abcdefgh>
```
Test to reproduce:
```
fail2ban-regex -l 5 -o msg '2022-10-04 10:31:49 imap-login: Info: Disconnected: Too many invalid commands (no auth attempts in 0 secs): user=<>, rip=192.0.2.111, lip=127.0.0.1, session=<abcdefgh>' dovecot
```
(failregex 1 needs few seconds to bypass)

WiP.

**Comment by foutrelis** [comment]
Thank you for the fix. It is much appreciated. :)

**Comment by pbobbenb** [comment]
Hi!

Glad someone else had the same problem, had just thought about reporting the issue when someone else beat me to it!

Had run f2b 0.11.2 for a long time without issue and upgraded to f2b 1.0.1 and it wouldn't last longer that 6-8h before the Python process hung at 100% cpu and no further f2b activity going on.
Tried downgrading f2b to 0.11.2 but keeping the newer filters, but same issue, wasn't fixed until I restored previous working filters.
Only use three filters, apache-modsecurity, dovecot and postfix.

I'm currently running on macOS 10.14.6 (Intel) with own compiled Python 3.8.14.

Kind regards
